File: 2D_amber/Assets/Scripts/CSharp/Character/DamageSystem.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class DamageSystem : MonoBehaviour {
	public float dmg = 1.0f ;
	public TargetType type = TargetType.Enemy ;
	
	void OnTriggerEnter2D(Collider2D co){
		switch((int)type){
			case 0:
				if(co.gameObject.layer == 11){
					co.GetComponent<Enemy>().Damage(dmg) ;
				}
				break ;
			case 1:
				if(co.gameObject.layer == 9){
					co.GetComponent<ControllerBattle>().Damage(dmg) ;
				}
				break ;
			case 2:
				if(co.gameObject.layer == 11 || co.gameObject.layer == 9){
					co.GetComponent<Enemy>().Damage(dmg) ;
				}
				break ;
		}
	}
	
	public enum TargetType{Enemy, Player, All} ;
}

File: 2D_amber/Assets/Scripts/JS/Character/DamageSystem.js
#pragma strict
var dmg : float = 1.0 ;
var type : TargetType = TargetType.Enemy ;

function OnTriggerEnter2D(co:Collider2D){
  
  switch(type){
    case 0:
      if(co.gameObject.layer == 11){
        co.GetComponent.<Enemy>().Damage(dmg) ;
      }
      break ;
    case 1:
      if(co.gameObject.layer == 9){
        co.GetComponent.<ControllerBattle>().Damage(dmg) ;
      }
      break ;
    case 2:
      if(co.gameObject.layer == 11 || co.gameObject.layer == 9){
      
      }
      break ;
  }
}

enum TargetType{Enemy, Player, All}
